plutus-core/plutus-ir/test/PlutusIR/Transform/DeadCode/Tests.hs

```diff
@@ -31,3 +35,14 @@ test_deadCode = runTestNestedIn ["plutus-ir/test/PlutusIR/Transform"] $
             , "recBindingComplex"
             , "pruneDatatype"
             ]
+
+test_typecheck :: TestTree
+test_typecheck = testGroup "typechecking"
+  [ testProperty "Builtin Variant1" $
+  withMaxSuccess 3000
+  (non_pure_typecheck_prop (removeDeadBindings (BuiltinsInfo DefaultFunSemanticsVariant1)))
+
+  , testProperty "Builtin Variant2" $
```

Comment:
I don't think this is really worth it. Which one we use is pretty irrelevant to the dead code pass

Reply:
Although I guess if you want you could just generate a random one!

Reply:
Funny enough the CI failed the second variant only (it passed locally). The only reason I'm not generating a random one is that I can't mark a specific variant as expected failure.

Reply:
I'm happy to just mark the whole thing as expected to fail in that case! A property test fails if you find a counterexample with some combination of random inputs. I don't see why the semantics variant should be different.

Reply:
I don't see why either. But remember we should throw away the assumption that we know what's happening! That's the whole reason we have tests! Yes it does make the code bigger this way but I think it makes sense to get as much fine grain detail as possible so we can debug more easily. With the random arguments all in one property tests, if it encounters one counter example it stops. Maybe I can use some of the functions in [this section](https://hackage.haskell.org/package/QuickCheck-2.14.3/docs/Test-QuickCheck.html#g:21)? It does seem wrong and silly to not use the random input feature but I feel I'm losing info/control if I do. @james-iohk any advice?

Reply:
I'm confused: what specifically do you think is going to happen here? I think: quickcheck will find the failing example and fail, thus proving a counterexample to the more-general property that this should work for all semantics variants. We can then investigate that failure.

Are you worried that the builtin semantics variant generator won't explore both options? Then you could use `label` or `cover` to check, but the generator is so simple I don't see the worry.  

> But remember we should throw away the assumption that we know what's happening! 

No, I'm saying that I know what is happening with _quickcheck_. I'm not saying I know what is happening with our code.

The whole point of quickcheck is to generate random arguments to functions and then run them. It's good at that, let's let it do its job!

> With the random arguments all in one property tests, if it encounters one counter example it stops.

So? We only need one counterexample at a time.

Reply:
>The whole point of quickcheck is to generate random arguments to functions and then run them. It's good at that, let's let it do its job!

I agree, hence my comment "It does seem wrong and silly to not use the random input feature"

Let's say we got a counter example, how do we start debugging? Do we know it fails because it's from a specific argument? If we test every argument separately, then we have some idea of that, because we'll know if e.g., the other variant or the other mode also fails or not. I was hoping I wouldn't lose this.

Perhaps I'm too worried about this. I'll just do the regular thing. People can do whatever they want to debug the failed tests.

Reply:
The counterexample will show the arguments, so you can just run it with those arguments to get the failure. And quickcheck will try and shrink them also. We might want to think about how we want the configuration arguments to shrink, but we can worry about that later.